.github/workflows/ci.yml

```diff
@@ -0,0 +1,67 @@
+---
+name: CI
+
+on:
+  push:
+    branches:
+      - main
+      - stable/*
+      - release-*
+  pull_request: {}
+  merge_group: {}
+  workflow_dispatch: {}
+
+concurrency:
+  cancel-in-progress: true
+  group: "${{ github.workflow }}-${{ github.ref }}"
+
+defaults:
+  run:
+    # use bash shell by default to ensure pipefail behavior is the default
+    # see https://docs.github.com/en/actions/using-workflows/workflow-syntax-for-github-actions#exit-codes-and-error-action-preference
+    shell: bash
+
+jobs:
+  detect-changes:
+    outputs:
+      actionlint: ${{ steps.filter.outputs.actionlint }}
+    runs-on: ubuntu-latest
+    timeout-minutes: 10
+    steps:
+      - uses: actions/checkout@v4
+      # Detect changes against the base branch
+      - name: Detect changes
+        uses: ./.github/actions/paths-filter
+        id: filter
+
+  actionlint:
+    if: needs.detect-changes.outputs.actionlint == 'true'
+    needs: [detect-changes]
+    runs-on: ubuntu-latest
+    timeout-minutes: 5
+    steps:
+      - uses: actions/checkout@v4
+      - run: echo "::add-matcher::.github/actionlint-matcher.json"
+      - run: |
+          bash <(curl https://raw.githubusercontent.com/rhysd/actionlint/main/scripts/download-actionlint.bash)
```

Comment:
💭 Feels a bit unsafe to do this. 

Maybe we can pin it to a version at least or use go to install it 🤔 We anyway need go then for the go client later? 🤔 https://github.com/rhysd/actionlint/blob/main/docs/install.md

WDYT?

Reply:
> we can pin it to a version at least

Probably makes sense but then also requires Renovate integration so the version is updated automatically so no humans needs to do that.

Reply:
Addressed that in the most recent commit :+1: